### PR TITLE
Add PyTorch module

### PR DESCRIPTION
**Pull request by suhacker1**
This PR allows users to fickle standard PyTorch files (files that have a `data.pkl`). It allows them to statically analyze, decompile, and inject code into this file. The current injection method is to combine the payload and original data into a PyTorch class with a custom `__reduce__` method. This resolves issue #29. 

I should also note that for proper torch importing, I changed `pickle.py` to `fickle.py`. I'm open to any suggestions for avoiding that given that this is a breaking change. 

As indicated by the title, this is a work-in-progress. The injection method needs to be refined (for fickling-native bytecode rewriting) as well as the PyTorch file format detector. I would appreciate feedback on the quality of the API in addition to the path manipulation and injection logic. In addition, I’m not satisfied with the test and example coverage of this feature as is. I also think the injection method should be analyzed further with regards to detectability. Another issue should be created to make sure all of these functionalities are also available from the CLI.

### CODE REVIEW

File: fickling/pytorch.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            injected_model = BaseInjection(self.pickled, payload)
+            torch.save(injected_model, output_path)
+        if overwrite == True:
+            # Rename the new file to replace the original file
+            Path(output_path).rename(self.path)
+            output_path = Path(self.output_path)
+            if output_path.exists():
+                os.remove(output_path)
+
+
```
Comment (ESultanik) [inline fickling/pytorch.py:67]: Is the code in the remainder of the file supposed to be here?

**Review by ESultanik — APPROVED** [review]
I'm fine with the refactor of `pickle` -> `fickle`; it's probably something we should have done from the get-go. But I worry that it might break other users' programmatic integrations with Fickling. Perhaps we should make this coincide with a major version jump to reinforce the breaking change?

There was also some code at the end of `pytorch.py` that looks like it was meant to have been deleted.

Other than that, this looks good to me!

File: fickling/pytorch.py (hunk 1 above)
Comment (suhacker1) [reply fickling/pytorch.py]: Removed

**Comment by suhacker1** [comment]
I've opted to split the aforementioned additions to the module for clarity.

**Comment by suhacker1** [comment]
I also think we should make this coincide with a major version jump to reinforce the breaking change.

**Pull request merged**